services/bonita-page/bonita-page-impl/src/main/java/org/bonitasoft/engine/page/impl/PageMappingServiceImpl.java

```diff
@@ -88,11 +88,19 @@ public void setAuthorizationRules(final List<AuthorizationRule> authorizationRul
 
     @Override
     public SPageMapping create(final String key, final Long pageId, final List<String> authorizationRules) throws SObjectCreationException {
-        final SPageMappingImpl entity = new SPageMappingImpl();
-        entity.setPageId(pageId);
-        entity.setPageAuthorizationRules(authorizationRules);
-        entity.setKey(key);
-        return insert(entity);
+        SPageMapping pageMapping = null;
+        try {
+            pageMapping = get(key);
+        } catch (final SObjectNotFoundException e) {
```

Comment:
Better not to create an exception for nothing:
Instead of doing get -> exception -> catch -> create  change the get or extract  submethod that do not throw exception
so you can do get -> if null -> create